src/shared/bus-unit-util.c

```diff
@@ -1415,6 +1418,32 @@ static int bus_append_execute_property(sd_bus_message *m, const char *field, con
                 return 1;
         }
 
+        if (streq(field, "RootHash")) {
+                _cleanup_free_ char *text = NULL;
+                _cleanup_free_ void *roothash_decoded = NULL;
+                size_t roothash_decoded_size = 0;
+
+                if (is_path(eq)) {
+                        /* We have the path to a roothash to load and decode, eg: RootHash=/foo/bar.roothash */
+                        r = read_one_line_file(eq, &text);
+                        if (r < 0)
+                                return r;
```

Comment:
nah, please don't read this client side. Client might live in different world than host. please pass this in a separate property RootHashPath= if it's a path